internal/gitmetadatabinary.go

```diff
@@ -0,0 +1,42 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2023 Datadog, Inc.
+
+//go:build go1.18
+// +build go1.18
+
+package internal
+
+import (
+	"runtime/debug"
+
+	"gopkg.in/DataDog/dd-trace-go.v1/internal/log"
+)
+
+// getTagsFromBinary extracts git metadata from binary metadata
+func getTagsFromBinary() map[string]string {
+	res := make(map[string]string)
+	info, ok := debug.ReadBuildInfo()
+	if !ok {
+		log.Warn("ReadBuildInfo failed, skip source code metadata extracting")
+		return res
+	}
+	goPath := info.Path
+	vcs := ""
```

Comment:
var vcs, commitSha string
```suggestion
	var vcs, commitSha string
```